internal/controller/postgrescluster/pki_test.go

```diff
@@ -44,10 +40,6 @@ import (
 )
 
 func TestReconcileCerts(t *testing.T) {
-	// Garbage collector cleans up test resources before the test completes
-	if strings.EqualFold(os.Getenv("USE_EXISTING_CLUSTER"), "true") {
-		t.Skip("USE_EXISTING_CLUSTER: Test fails due to garbage collection")
-	}
```

Comment:
I think we want this still so that this test won't run at all in our existing cluster tests (and then later on, we _should_ have a ticket for removing all these guards and removing all notion of existing cluster tests from our makefile, github actions, etc.).

Reply:
Yeah, I considered this in general, but neglected to consider our github actions, that may be an issue. I'll add this back and put a note on the current cleanup story (let me know if you think it should be split out in a separate one).

Reply:
I think it would be nice to do it all in one ticket and then when someone asks we can say "envtest-existing? doesn't sound familiar."

Reply:
Actually, now I'm second-guessing myself about these skips if `USE_EXISTING_CLUSTER` is true: 
* on one hand, we will not be using an existing cluster, so it makes sense to remove these guard-rails
* on the other hand, if someone does run the test with USE_EXISTING_CLUSTER=true, these guard-rails will prevent the tests from erroring out when there's no real error

What do y'all think?